### PR TITLE
Limit secret requests when listing app repositories

### LINKED ISSUE #1686: Avoid requesting secrets in a namespace
### Description:

In order to be able to update the existing AppRepositories and to be able to select Docker credentials, we are listing all the secrets in a namespace in the Dashboard. We should be able to limit the info requested in order to avoid potential security issues.

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

We were requesting all the secrets in a namespace when listing the app repositories. This was for later on, being able to render secrets associated to a repo or to select some docker registry credentials. This PR changes that:

 - The request is now done only when rendering the new/edit repo form.
 - We are listing only docker registry secrets
 - We are getting just the secret associated to a repository (when editing it).

### Applicable issues

  - fixes #1686 

### Additional information

If you request some changes, I will eventually come back to this and address your comments :)

### CODE REVIEW

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx [hunk 1]
```diff
@@ -50,6 +50,12 @@ export function AppRepoAddDockerCreds({
   const handleServerChange = (e: React.ChangeEvent<HTMLInputElement>) => setServer(e.target.value);
   const toggleCredSubForm = () => setShowSecretSubForm(!showSecretSubForm);
 
+  useEffect(() => {
+    if (imagePullSecrets.length && !currentImagePullSecrets.length) {
```
Comment (absoludity) [inline dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx:54]: This condition appears to only be true if there were no image pull secrets previously? What if a single secret is changed, or a secret is added (not sure if that's supported, but if not, a comment justifying the simpler condition would help), or if a secret is removed? Seems in those cases we wouldn't be setting the current pull secrets?

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx [hunk 2]
```diff
@@ -208,7 +202,7 @@ function AppRepoList() {
               <LoadingWrapper
                 className="margin-t-xxl"
                 loadingText="Fetching Application Repositories..."
-                loaded={!isFetching}
+                loaded={!isFetchingElem.repositories}
```
Comment (absoludity) [inline dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:205]: Not clear to me why this change is here? Looks like we're making the condition for loaded be more specific, to only depend on the repositories?

**Review by absoludity — APPROVED** [review]

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx [hunk 3]
```diff
@@ -104,30 +108,43 @@ export function AppRepoForm(props: IAppRepoFormProps) {
...
-          } else {
-            setAuthMethod(AUTH_METHOD_CUSTOM);
-            setAuthHeader(atob(secret.data.authorizationHeader));
-          }
+        if (repo.spec?.auth?.header && !secrets.includes(repo.spec.auth.header.secretKeyRef.name)) {
+          secrets.push(repo.spec.auth.header.secretKeyRef.name);
         }
-        if (secret.data[".dockerconfigjson"]) {
-          setAuthMethod(AUTH_METHOD_REGISTRY_SECRET);
+        secrets.forEach(s => dispatch(actions.repos.fetchRepoSecret(namespace, s)));
```
Comment (antgamdia) [inline dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:119]: Mmm, interesting, so in the AppRepoForm, once loaded, we are fetching:

1. The whole list of `imagePullSecrets` un a given namespace
2. The `repoSecret` each time a repo is used/selected/etc

no?

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx (hunk 1 above)
Comment (antgamdia) [reply dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx:54]: So perhaps we should compare both objects and merge them, instead of replacing it the `currentImagePullSecrets` is empty?

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx (hunk 2 above)
Comment (antgamdia) [reply dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:205]: Are we using this `isFetchingElem` elsewhere? I don't see any `isFetchingElem.secrets` for example in the code, for example.
Just out of curiosity, I'm keen to have more granularity here.

**Review by antgamdia — APPROVED** [review]

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx (hunk 2 above)
Comment (absoludity) [reply dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:205]: > Looks like we're making the condition for loaded be more specific, to only depend on the repositories?

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx (hunk 3 above)
Comment (absoludity) [reply dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:119]: On another tangent, this is a good example where an API for fetching secrets *without* the actual secret data being sent over the wire would be useful, IMO, and only fetching the actual secret data when required for a particular secret. But that's a goal for the new API design, not this PR :)

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx (hunk 1 above)
Comment (andresmgot) [reply dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx:54]: no, this is just the `useEffect` of the component, the safe-guard is to avoid re-setting in the component state `currentImagePullSecrets` (and force a re-render) if it's been already set.

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx (hunk 2 above)
Comment (andresmgot) [reply dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:205]: Yes, this is because before we were fetching secrets and repos here (now just repos). Without this change, the apprepo form modal is closed just after opening it. This is because we are requesting secrets at that moment and the parent component, this loadingWrapper, was detecting the change and rendering the loading gif instead.

**Review by andresmgot — COMMENTED** [review]
Had some time to go back to this :)
